wire/internal/wire/testdata/UnusedInputs/foo/wire.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build wireinject
+
+package main
+
+import (
+	"github.com/google/go-cloud/wire"
+)
+
+func injectBar() Bar {
+	wire.Build(
+		provideFoo,                            // needed as input for provideBar
+		provideBar,                            // needed for Bar
+		provideUnused,                         // not needed -> error
+		wire.Value("unused"),                  // not needed -> error
+		wire.NewSet(provideUnusedInSet),       // nothing in set is needed -> error
```

Comment:
Nice, this covers this quite thoroughly! Two more cases to check:

1. A set that is partially used (i.e. contains at least one provider that is used and at least one provider that is not) should not produce an error.
2. A set that is fully used should not produce an error.